cli/utils.go

```diff
@@ -207,3 +209,47 @@ func writeJSON(cmd *cobra.Command, out any) error {
 	enc.SetIndent("", "  ")
 	return enc.Encode(out)
 }
+
+// WithExampleRegistry injects an ExampleRegitry into the context.
+// This is primarily only needed by the tests but needs to exist
+// in the main CLI package
+func WithExampleRegistry(ctx context.Context, registry *ExampleRegistry) context.Context {
+	return context.WithValue(ctx, exampleRegistryCtxKey{}, registry)
+}
+
+type ExampleRegistry struct {
+	examples map[string]string
+}
+
+func EmbedCLIExample(ctx context.Context, cmd *cobra.Command, name, usage string) {
```

Comment:
todo: Some documentation on this will be nice.

Mostly because this is public, and I can see being used quite a lot in the future.